### PR TITLE
Feature/3045/starred organization events

### PR DESCRIPTION
**Pull request by garyluu**
For partial #3045 , UI remains

Add query for events related to starred organization (unused)

Add query for events related to starred organization OR starred entries

The previous query for events related to starred entries is also unused

Also need to remove all swagger annotations from the eventresource

### CODE REVIEW

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java  3
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DAGHelper.java  100
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java  88
+ dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java  34
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java  58
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java  11
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java  44
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java  17
+ dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java  82
+ dockstore-common/src/main/java/io/dockstore/common/NextflowUtilities.java  73
+ dockstore-webservice/src/main/java/io/dockstore/webservice/core/dag/ElementsDefinition.java  100
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryLabelHelper.java  5
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java  20
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java  82
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java  2
+ dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java  11
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java  16
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java  1
+ dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoFactory.java  2
         

Coverage decreased per file
===========================
- dockstore-webservice/src/main/java/io/swagger/model/ExtendedFileWrapper.java  -8
- dockstore-common/src/main/java/io/dockstore/common/Repository.java  -56
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Token.java  -9
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/dockerhub/DockerHubTag.java  -15
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java  -34
- dockstore-common/src/main/java/io/dockstore/common/Utilities.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/User.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/StarrableResourceInterface.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/Permission.java  -63
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryStarredSerializer.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DockstoreYaml.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java  -18
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSEntry.java  -79
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java  -5
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ServiceDAO.java  -40
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionResource.java  -69
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ManualRegistry.java  -77
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/BioWorkflowDAO.java  -50
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSFeed.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/OrganizationUser.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ApiVersionConverter.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java  -6
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/RSSListener.java  -30
- dockstore-webservice/src/main/java/io/swagger/model/MetadataV1.java  -48
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  -19
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedToolResource.java  -59
- dockstore-common/src/main/java/io/dockstore/common/EntryUpdateTime.java  -55
- dockstore-webservice/src/main/java/io/swagger/model/ToolVersionV1.java  -52
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/Config.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/NotificationResource.java  -46
- dockstore-webservice/src/main/java/io/swagger/model/ToolDescriptor.java  -44
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java  -36
- dockstore-webservice/src/main/java/io/swagger/model/ToolV1.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Organization.java  -76
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsExtendedApi.java  -33
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSHeader.java  -81
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java  -31
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Workflow.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/PublicStateManager.java  -15
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ImageRegistryFactory.java  -42
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/CollectionOrganization.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/WorkflowPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java  -93
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsInterface.java  -79
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/SharedWorkflows.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java  -20
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java  -19
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java  -56
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/RSSWorkflowPath.java  -100
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolClassesApiServiceImpl.java  -45
- dockstore-webservice/src/main/java/io/swagger/model/ToolTestsV1.java  -24
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java  -19
- dockstore-webservice/src/main/java/io/dockstore/webservice/languages/ServicePrototypePlugin.java  -64
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java  -54
- dockstore-common/src/main/java/io/dockstore/common/Registry.java  -12
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/ToolDAO.java  -8
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TokenDAO.java  -21
- dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java  -3
- dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java  -61
- dockstore-webservice/src/main/java/io/swagger/api/MetadataApiV1.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/MetadataResourceHelper.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/NotificationDAO.java  -85
- dockstore-common/src/main/java/io/dockstore/common/OrganizationUpdateTime.java  -57
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java  -2
- dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java  -10
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/UserDAO.java  -29
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java  -67
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/Limits.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/TRSListener.java  -17
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java  -45
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java  -2
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java  -18
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/RSSToolPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/rss/RSSWriter.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java  -79
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java  -29
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/ToolPath.java  -100
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZipSourceFileHelper.java  -65
- dockstore-webservice/src/main/java/io/swagger/api/ToolClassesApiV1.java  -34
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java  -48
- dockstore-webservice/src/main/java/io/swagger/api/impl/MetadataApiServiceImpl.java  -89
- dockstore-webservice/src/main/java/io/dockstore/webservice/SimpleAuthorizer.java  -50
- dockstore-webservice/src/main/java/io/swagger/api/ToolsApiV1.java  -87
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/ExtendedUserData.java  -72
- dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java  -9
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Event.java  -22
- dockstore-webservice/src/main/java/io/swagger/model/ToolDockerfile.java  -23
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java  -37
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Service.java  -43
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/TokenType.java  -7
- dockstore-webservice/src/main/java/io/dockstore/webservice/api/StarRequest.java  -62
- dockstore-webservice/src/main/java/io/dockstore/webservice/core/Notification.java  -54
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java  -11
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/OrganizationDAO.java  -67
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ServiceResource.java  -60
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java  -27
- dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/SitemapListener.java  -25
- dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/CollectionDAO.java  -59
         

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java [hunk 1]
```diff
@@ -68,20 +72,32 @@ public EventResource(EventDAO eventDAO, UserDAO userDAO) {
...
-    public List<Event> getEvents(@ApiParam(hidden = true) @Auth User user, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
-        if (eventSearchType.equals(EventSearchType.STARRED_ENTRIES)) {
-            User userWithSession = this.userDAO.findById(user.getId());
+    public Set<Event> getEvents(@Parameter(hidden = true) @ApiParam(hidden = true) @Auth User user, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java]: Why did you change this from a List to a Set? Won't you lose the queries' sort orders?

**Comment by garyluu** [comment]
PR passed, will fix merge conflicts after review

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java (hunk 1 above)
Comment (garyluu) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java]: Was originally attempting to remove duplicate events, I'll change it back to list.  I don't think sorted set works anyways.

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

File: dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java [hunk 2]
```diff
@@ -331,6 +334,31 @@ public void testCreateNewOrganization() {
...
+        testStarredOrganizationEvents(organizationsApiUser2, organization);
+    }
+
+    private void testStarredOrganizationEvents(OrganizationsApi organizationsApiUser2, Organization organization) {
+        organizationsApiUser2.starOrganization(organization.getId(), SwaggerUtility.createStarRequest(true));
+        final io.dockstore.openapi.client.ApiClient openAPIWebClientUser2 = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        EventsApi eventsApi = new EventsApi(openAPIWebClientUser2);
+        List<io.dockstore.openapi.client.model.Event> events = eventsApi
+                .getEvents(EventSearchType.STARRED_ORGANIZATION.toString(), null, null);
+        Assert.assertEquals("Should have 1 create, 1 approve, 4 modify events", 6, events.size());
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java]: A bit misleading, the test only checks that there are 6 events in general. 
Same idea with two lines down

Comment (garyluu) [reply dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java]: It was only supposed to test for 6 events.  The comment was only to provide additional details for debugging purposes.

Comment (denis-yuen) [reply dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java]: Understood, what I meant was that the test would be better if it actually checked for 1 create, 1 approve, 4 modify events. 
Right now, it looks like the test could pass if it got 6 create events.

Comment (garyluu) [reply dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java]: Yes, that's what it's intended to do.  I'm testing whether the pagination works for this endpoint not whether the event type was correctly created.  We have other tests for that.

**Comment by denis-yuen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dockstore/dockstore/pullRequest?prid=4793096&bid=15956691)

**Comment by garyluu** [comment]
> It should actually say 6 events so that the debugger knows the target, but close enough

Oh, I intentionally remove that because on assertion error, the output is something like "expected 6 but got 7"
